File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        obj = {')': '(', '}': '{', ']': '['}

        for c in list(s):
            if c in '({[':
                stack.append(c)
            else:
                if not len(stack) or stack.pop() != obj[c]:
                    return False
        
        return len(stack) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const obj = {'(': ')', '{': '}', '[': ']'};
    const stack = [];
    for (const c of Array.from(s)) {
        if (c in obj) {
            stack.push(c);
        } else {
            if (!stack.length) return false;
            const key = stack.pop();
            if (obj[key] !== c) return false;
        }
    }
    return stack.length === 0;
    
};